### PR TITLE
Old shool processing

### PR DESCRIPTION
**Pull request by jebuss**
* added functionality to process data the good old way, with qsub
* moved conversion of fact_tools json output to hdf5 into a dedicated funciton
* facttools_executer: Excuter script on workernodes for qsub batch processing
* process_fact_data_qsub: script to submit fact_tools jobs to the batch system
* changed process_fact_data in order to define the number of runs per requested jobs instead of the number of jobs in total

### CODE REVIEW

File: erna/scripts/facttools_executer.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    '''
+    This is what will be executed on the cluster
+    '''
+    logger = logging.getLogger(__name__)
+    logger.info("facttools executor has been started.")
+
+    input_path = os.environ.get("INPUTFILE")
+    jar = os.environ.get("JARFILE")
+    xml = os.environ.get("XMLFILE")
+    output_path = os.environ.get("OUTPUTFILE")
```
Comment (mackaiver) [inline erna/scripts/facttools_executer.py]: Cant we use tempfiles here? The can clean up after themselves which is nice.

File: setup.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
         'erna': ['resources/*'],
     },
     install_requires=[
+        'dask>=0.9'         # in anaconda
```
Comment (mackaiver) [inline setup.py]: where do you need the dask stuff?

Comment (jebuss) [inline setup.py]: ups, not yet but hopefully soon

File: erna/__init__.py [hunk 3]
```diff
@@ -200,3 +202,19 @@ def load(earliest_night, latest_night, path_to_data, factdb,source_name="Crab",
...
+
+def ft_json_to_df(json_path):
+    with open(json_path,'r') as text:
+        try:
+            logger.info("Reading fact-tools output.")
+            y=json.loads(text.read())
+            df_out=pd.DataFrame(y)
+            logger.info("Returning data frame with {} entries".format(len(df_out)))
+            return df_out
+        except ValueError as e:
```
Comment (MaxNoe) [inline erna/__init__.py]: You should use logger.exception here. This prints the complete traceback

File: erna/__init__.py [hunk 4]
```diff
@@ -200,3 +202,19 @@ def load(earliest_night, latest_night, path_to_data, factdb,source_name="Crab",
...
+            logger.error("Fact-tools output could not be read.")
+            print(e)
```
Comment (MaxNoe) [inline erna/__init__.py]: No prints

File: erna/qsub.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        df = pd.read_csv(ret.stdout, delimiter="\s+")
+        # drop the first line since it is just one long line
+        df = df.drop(df.index[0]).copy()
+        # convert objects to numeric otherwise numbers are strings
+        df["JOBID"] = pd.to_numeric(df["job-ID"], errors='coerce')
+        # df.set_index("JOBID")
+        df = df.drop('job-ID', 1)
+
+    except ValueError as e:
+        print("No jobs in queues for user {}".format(user), e)
```
Comment (MaxNoe) [inline erna/qsub.py]: Use logging instead of print

File: erna/scripts/facttools_executer.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    try:
+        df_out = ft_json_to_df(json_output_path)
+        df_out["fact_tools"] = os.path.basename(jar)
+        df_out["xml"] = os.path.basename(xml)
+        df_out.to_hdf(os.path.abspath(hdf_output_path), 'data', mode='w')
+        os.remove(input_path)
+        os.remove(json_output_path)
+    except Exception as e:
+        logger.error("Fact-tools output could not be read.")
```
Comment (MaxNoe) [inline erna/scripts/facttools_executer.py]: logger.exception

File: erna/scripts/process_fact_data_qsub.py [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+
+
+def read_outputs_to_list(job_output_paths):
+    job_outputs = []
+    for job_output_path in job_output_paths:
+        try:
+            df_out = pd.read_hdf(job_output_path, "data")
+            job_outputs.append(df_out)
+        except Exception as e:
+            logger.error("{} could not be appended.".format(job_output_path))
```
Comment (MaxNoe) [inline erna/scripts/process_fact_data_qsub.py]: logging.exception

**Comment by jebuss** [comment]
Soooo, worked in all comments. If you don't have any complaints I will merge. +/-1 me!

**Comment by MaxNoe** [comment]
nice

**Pull request merged**